test/stablecoin_converter.js

```diff
@@ -223,6 +223,186 @@ contract("StablecoinConverter", async (accounts) => {
+      const stablecoinConverter = await StablecoinConverter.new(2 ** 16 - 1)
+      const erc20_1 = await MockContract.new()
+      const erc20_2 = await MockContract.new()
+
+      await erc20_1.givenAnyReturnBool(true)
+      await erc20_2.givenAnyReturnBool(true)
+
+      await stablecoinConverter.deposit(erc20_1.address, 10, { from: user_1 })
+      await stablecoinConverter.deposit(erc20_2.address, 20, { from: user_2 })
+
+      await stablecoinConverter.addToken(erc20_1.address)
+      await stablecoinConverter.addToken(erc20_2.address)
+      const batchIndex = (await stablecoinConverter.getCurrentStateIndex.call()).toNumber()
+
+      const orderId1 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 1, 0, true, batchIndex + 1, 20, 10, { from: user_1 })
+      const orderId2 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 0, 1, true, batchIndex + 1, 10, 20, { from: user_2 })
+      // close auction
+      await waitForNSeconds(BATCH_TIME)
+
+      const prices = [10, 20]
+      const owner = [user_1, user_2]  //tradeData is submitted as arrays
+      const orderId = [orderId1, orderId2]
+      const volume = [5, 10]
+
+      const tokenIdForPrice = [0, 1]
+
+      await stablecoinConverter.submitSolution(batchIndex, owner, orderId, volume, prices, tokenIdForPrice)
+
+      assert.equal((await stablecoinConverter.getBalance.call(user_1, erc20_1.address)).toNumber(), 5, "Sold tokens were not adjusted correctly")
+      assert.equal(await stablecoinConverter.getBalance.call(user_1, erc20_2.address), 10, "Bought tokens were not adjusted correctly")
+      assert.equal(await stablecoinConverter.getBalance.call(user_2, erc20_1.address), 5, "Bought tokens were not adjusted correctly")
+      assert.equal((await stablecoinConverter.getBalance.call(user_2, erc20_2.address)).toNumber(), 10, "Sold tokens were not adjusted correctly")
+
+      const volume2 = [10, 20]
+
+      await stablecoinConverter.submitSolution(batchIndex, owner, orderId, volume2, prices, tokenIdForPrice)
+
+      assert.equal((await stablecoinConverter.getBalance.call(user_1, erc20_1.address)).toNumber(), 0, "Sold tokens were not adjusted correctly")
+      assert.equal(await stablecoinConverter.getBalance.call(user_1, erc20_2.address), 20, "Bought tokens were not adjusted correctly")
+      assert.equal(await stablecoinConverter.getBalance.call(user_2, erc20_1.address), 10, "Bought tokens were not adjusted correctly")
+      assert.equal((await stablecoinConverter.getBalance.call(user_2, erc20_2.address)).toNumber(), 0, "Sold tokens were not adjusted correctly")
+    })
+    it("checks solution submission with 3 different solutions", async () => {
+      const stablecoinConverter = await StablecoinConverter.new(2 ** 16 - 1)
+      const erc20_1 = await MockContract.new()
+      const erc20_2 = await MockContract.new()
+
+      await erc20_1.givenAnyReturnBool(true)
```

Comment:
I'd be ok with having a utility method that registers n tokens (since it's used in all tests).

Reply:
let me do this in the next PR. I have already rewritten it in there with a fee token and the merge would be quite complicated, but yeah, good comment